src/include/common/Histogram.h

```diff
@@ -0,0 +1,429 @@
+#include <queue>
+#include <shared_mutex>
+#include <thread>
+#include <unordered_map>
+
+#include "TDigest.h"
+
+#define MIN_IN_MILLI 60000     /* Define 1 minute in millisecond */
+
+
+namespace wavefront {
+
+/* Wrapper for TDigest distribution */
+class Snapshot {
+  public:
+    /**
+     * @description: constructor of Snapshot
+     * @param tdigest::TDigest
+     * @return: tdigest::TDigest
+     */
+    Snapshot(tdigest::TDigest t) { snapshot_digest = t; }
+
+    Snapshot(const Snapshot &o) { snapshot_digest = o.snapshot_digest; }
+    /**
+     * @description: get max value in snapshot
+     * @return: max value in snapshot
+     */
+    double GetMax() { return snapshot_digest.quantile(1.0); }
+
+    /**
+     * @description: get min value in snapshot
+     * @return: minimum value in snapshot
+     */
+    double GetMin() { return snapshot_digest.quantile(0.0); }
+
+    /**
+     * @description: get mean value in snapshot
+     * @return: mean value in snapshot
+     */
+    double GetMean() { return snapshot_digest.getMean(); }
+
+    /**
+     * @description: get sum of snapshot.
+     * @return: sum in snapshot.
+     */
+    double GetSum() { return snapshot_digest.getSum(); }
+
+    /**
+     * @description: get total weight of snapshot.
+     * @return: total weight in snapshot.
+     */
+    double GetWeight() { return snapshot_digest.getWeight(); }
+
+    /**
+     * @description: get # of centroids of snapshot.
+     * @return: total # of centroids in snapshot.
+     */
+    int GetCount() { return snapshot_digest.totalSize(); }
+
+    /**
+     * @description: get the value of given quantile.
+     * @param quantile, range between 0.0 and 1.0.
+     * @return: the value in the distribution at the given quantile, or NAN if
+     * snapshot is empty or param out of range.
+     */
+    double GetValue(double quantile) {
+        return snapshot_digest.quantile(quantile);
+    }
+
+    tdigest::TDigest snapshot_digest;
+};
+
+class Distribution {
+    /**
+     * Representation of a histogram distribution.
+     *Containing a timestamp and a list of centroids.
+     */
+  public:
+    Distribution(){};
+
+    /**
+     * @description: Construct a distribution.
+     * @param timestamp: Timestamp in milliseconds since the epoch.
+     * @param centroids: Centroids
+     */
+    Distribution(uint64_t timpstamp_,
+                 std::vector<std::pair<double, double>> centroids_) {
+        timestamp = timpstamp_;
+        centroids = centroids_;
+    }
+
+    Distribution(const Distribution &o) {
+        timestamp = o.timestamp;
+        centroids = o.centroids;
+    }
+
+    uint64_t timestamp;
+    std::vector<std::pair<double, double>> centroids;
+};
+
+class ThreadMinBin {
+  public:
+    ThreadMinBin() {}
+    /**
+     * @description: Construct a ThreadMinBin.
+     * @param _accuary: accuary
+     *        _minute_millis: minute in millisecond
+     */
+    ThreadMinBin(double _accuary, uint64_t _minute_millis) {
```

Comment:
Follow the common practice of C++ constructor, be consistent with ones in `TDigest.h`. Same for the rest of the PR.